### DISCUSSION TITLE
Predictable package urls

### DISCUSSION THREAD

**Discussion by davhdavh** [community/community#69164] [Packages]
### Select Topic Area

Product Feedback

### Body

Current url to package release is https://github.com/xxx/yyy/pkgs/nuget/ppp/nnn with xxx/yyy being the standard org/repo and ppp is the nuget package name, and yyy is some id.

Could you please support https://github.com/xxx/yyy/pkgs/nuget/ppp/vvv where vvv is the release version, e.g. `1.3.0`.

As it is right now, it is impossible to link a package in a release.

**Comment by esurov** [comment]
Dear @davhdavh ,

I can't directly make changes to GitHub or its features, but I can offer some guidance on how you might be able to achieve your goal using the existing GitHub features. To link to a specific package release version on GitHub, you can follow these steps:

1. Publish a Release:
   Make sure you have a release for your repository. If you don't have one, create a release by going to your GitHub repository, clicking on the "Releases" tab, and then clicking "Draft a new release."

2. Include Package Information:
   In your release notes or description, include the information about the NuGet package version, such as "NuGet Package v1.3.0." This will help users understand which package version is associated with this release.

3. Use GitHub Release URL:
   GitHub provides a URL for each release you create. The URL follows the format `https://github.com/xxx/yyy/releases/tag/vvv`, where `vvv` is the release version you specified. You can link directly to this URL to provide a specific version of your package.

For example, if your release version is 1.3.0 and your GitHub repository is `https://github.com/xxx/yyy`, you can link to the release like this:
`https://github.com/xxx/yyy/releases/tag/1.3.0`

This way, users can access the specific release, and in your release description, you can provide additional information about the package version and any other release notes or changes.

Remember that the specific URLs you mentioned (`https://github.com/xxx/yyy/pkgs/nuget/ppp/nnn` and `https://github.com/xxx/yyy/pkgs/nuget/ppp/vvv`) may not be standard GitHub features, and you might need to rely on the "Releases" section for linking to specific release versions.

**Reply by davhdavh** [reply]
Hi @esurov 
The problem is not making a release, the problem is that adding a link to a package in the release is impossible without lots of manual work.

A reasonable deployment strategy will look at the tag generated by making a new release and then do the actual build and upload of packages to various places. IE, the actual release happens AFTER the GitHub release. In this scenario you have no idea what the direct link is to the package. You CAN go back and find the link(s) after package upload and then paste the links in then, but again manual work.

Another reasonable deployment strategy is calculate the next release version based on commit messages (semantic commit logs), and then upload the packages and then generate the release. But again here you have lost, because you cannot get the url of the uploaded package in a standard way, and thus have to revert to manual work.
